crates/stackable-operator/src/kvp/mod.rs

```diff
@@ -90,265 +94,102 @@ where
-    T: Value,
-{
-    type Target = BTreeSet<KeyValuePair<T>>;
-
-    fn deref(&self) -> &Self::Target {
-        &self.0
-    }
-}
-
-impl<T> KeyValuePairs<T>
-where
-    T: Value + std::default::Default,
-{
-    /// Creates a new empty list of [`KeyValuePair`]s.
-    pub fn new() -> Self {
-        Self::default()
-    }
-
-    /// Creates a new list of [`KeyValuePair`]s from `pairs`.
-    pub fn new_with(pairs: BTreeSet<KeyValuePair<T>>) -> Self {
-        Self(pairs)
-    }
-
-    /// Extends `self` with `other`.
-    pub fn extend(&mut self, other: Self) {
-        self.0.extend(other.0);
-    }
-
-    /// Inserts a new [`KeyValuePair`] into the list of pairs.
-    ///
-    /// This function overwrites any existing key/value pair. To avoid
-    /// overwriting existing pairs, either use [`KeyValuePairs::contains`] or
-    /// [`KeyValuePairs::contains_key`] before inserting or try to insert
-    /// fallible via [`KeyValuePairs::try_insert`].
-    pub fn insert(&mut self, kvp: KeyValuePair<T>) -> &mut Self {
-        self.0.insert(kvp);
-        self
-    }
-
-    /// Tries to insert a new [`KeyValuePair`] into the list of pairs.
-    ///
-    /// If the list already had this pair present, nothing is updated, and an
-    /// error is returned.
-    pub fn try_insert(&mut self, kvp: KeyValuePair<T>) -> Result<(), KeyValuePairsError> {
-        ensure!(!self.0.contains(&kvp), PairAlreadyExistsSnafu);
-        self.insert(kvp);
-        Ok(())
-    }
-
-    /// Returns if the list contains a specific [`KeyValuePair`].
-    pub fn contains(&self, kvp: impl TryInto<KeyValuePair<T>>) -> bool {
-        let Ok(kvp) = kvp.try_into() else {
-            return false;
-        };
-        self.0.contains(&kvp)
-    }
 
-    /// Returns if the list contains a key/value pair with a specific [`Key`].
-    pub fn contains_key(&self, key: impl TryInto<Key>) -> bool {
-        let Ok(key) = key.try_into() else {
+    fn contains_str_key(&self, key: &str) -> bool {
+        // We could avoid this clone by providing an UnvalidatedKeyRef and ensure that Key: Borrow<UnvalidatedKeyRef>
+        let Ok(key) = key.parse::<Key>() else {
+            // If the key cannot be parsed then it cannot, by definition, possibly exist in the map
             return false;
         };
-
-        for kvp in &self.0 {
-            if kvp.key == key {
-                return true;
-            }
-        }
-
-        false
-    }
-
-    /// Returns an [`Iterator`] over [`KeyValuePairs`] yielding a reference to every [`KeyValuePair`] contained within.
-    pub fn iter(&self) -> impl Iterator<Item = &KeyValuePair<T>> {
-        self.0.iter()
+        self.contains_key(&key)
     }
 }
 
-impl<T> IntoIterator for KeyValuePairs<T>
-where
-    T: Value,
-{
-    type Item = KeyValuePair<T>;
-    type IntoIter = std::collections::btree_set::IntoIter<Self::Item>;
+impl<'a, T: Value> TryFromIterator<(&'a str, &'a str)> for KeyValuePairs<T> {
+    type Error = KeyValuePairError<T::Error>;
```

Comment:
nit: Use `V` instead of `T` to be consistent.

```suggestion
impl<'a, V: Value> TryFromIterator<(&'a str, &'a str)> for KeyValuePairs<V> {
    type Error = KeyValuePairError<V::Error>;
```

Reply:
17458e1afd86e674a63cd99c3fabcc3261786192